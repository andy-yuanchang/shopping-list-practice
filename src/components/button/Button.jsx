import React from 'react';
import PropTypes from 'prop-types';

import './Button.less';

function Button(props) {
  const { text, onClick, size = 'medium' } = props;

  return (
    <div
      className={`button ${size}`}
      onClick={onClick}
      onKeyDown={undefined}
      role="button"
      tabIndex={0}
      aria-label="button"
    >
      {text}
    </div>
  );
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  size: PropTypes.string.isRequired,
};

export default Button;
