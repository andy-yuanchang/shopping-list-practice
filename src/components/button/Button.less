@import '../generalAttribute.less';

.button {
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1rem;
  font-weight: 700;
  color: white;
  background: @main-color;
  cursor: pointer;
  width: 4rem;
  height: 2rem;
  &:hover {
    transform: scale(1.05);
    transition: transform 0.1s ease-out;
    box-shadow: 2px 2px 2px 1px rgba(0, 0, 0, 0.2);
  }
}