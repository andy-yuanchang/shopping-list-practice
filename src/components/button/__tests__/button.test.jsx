import React from 'react';
import { render, cleanup } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';

import Button from '../Button';

afterEach(cleanup);

describe('Button', () => {
  it('order text small size', () => {
    const button = TestRenderer.create(
      <Button
        text="order"
        onClick={() => {}}
        size="small"
      />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });

  it('order text medium size', () => {
    const button = TestRenderer.create(
      <Button
        text="order"
        onClick={() => {}}
        size="medium"
      />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });

  it('order text large size', () => {
    const button = TestRenderer.create(
      <Button
        text="order"
        onClick={() => {}}
        size="large"
      />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });
});
