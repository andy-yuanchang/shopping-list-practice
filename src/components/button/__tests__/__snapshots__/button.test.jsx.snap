// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button order text large size`] = `
<div
  className="button large"
  onClick={[Function]}
>
  order
</div>
`;

exports[`Button order text large size 1`] = `
<div
  className="button large"
  onClick={[Function]}
>
  order
</div>
`;

exports[`Button order text medium size`] = `
<div
  className="button medium"
  onClick={[Function]}
>
  order
</div>
`;

exports[`Button order text medium size 1`] = `
<div
  className="button medium"
  onClick={[Function]}
>
  order
</div>
`;

exports[`Button order text small size`] = `
<div
  className="button small"
  onClick={[Function]}
>
  order
</div>
`;

exports[`Button order text small size 1`] = `
<div
  className="button small"
  onClick={[Function]}
>
  order
</div>
`;

exports[`Button undefined large size`] = `
<div
  className="button large"
  onClick={[Function]}
>
  undefined
</div>
`;

exports[`Button undefined large size 1`] = `
<div
  className="button large"
  onClick={[Function]}
>
  undefined
</div>
`;

exports[`Button undefined medium size`] = `
<div
  className="button medium"
  onClick={[Function]}
>
  undefined
</div>
`;

exports[`Button undefined medium size 1`] = `
<div
  className="button medium"
  onClick={[Function]}
>
  undefined
</div>
`;

exports[`Button undefined small size`] = `
<div
  className="button small"
  onClick={[Function]}
>
  undefined
</div>
`;

exports[`Button undefined small size 1`] = `
<div
  className="button small"
  onClick={[Function]}
>
  undefined
</div>
`;
